webapp/src/utils/jira_issue_metadata.tsx

```diff
@@ -212,7 +214,19 @@ export function getCustomFieldFiltersForProjects(metadata: IssueMetadata | null,
         } as FilterField;
     });
 
-    const result = populatedFields.concat(userDefinedFields);
+    const commentVisibility = selectFields.map((field) => {
+        return {
+            key: commentVisibilityFieldKey,
+            name: 'Comment Visibility',
+            schema: {
+                type: 'commentVisibility',
+            },
+            values: [],
+            issueTypes: field.validIssueTypes,
+        } as FilterField;
+    });
+
+    const result = populatedFields.concat(userDefinedFields, commentVisibility);
```

Comment:
I think this `selectFields.map` is why we see several "Comment Security" objects in the jest snapshot

Will this work:

```ts
const issueTypes: IssueTypeIdentifier[] = metadata.projects[0].issuetypes.map(({id, name}) => ({id, name}));

const commentVisibility = {
    key: commentVisibilityFieldKey,
    name: 'Comment Visibility',
    schema: {
        type: 'commentVisibility',
    },
    values: [],
    issueTypes,
} as FilterField;
```